### DISCUSSION TITLE
How to handle single data points in pushgateway/grafana

### DISCUSSION THREAD

**Discussion by vegetablesalad** [prometheus/pushgateway#430] [Q&A]
Hello. Need some advice.
I have a script that sends data to push-gateway on different events caused by some user action.
It simple numeric data 0,1,2,3,4,5.
This is how my data looks in grafana:

![Screenshot 2021-10-05 at 11 36 05](https://user-images.githubusercontent.com/2006277/135990435-dbfd614b-0929-49f0-84fd-462bb2bd21d1.png)

The problem here is that I send one data point to pushgateway and prometheous/grafana will now stretch that datapoint until it receives a new one. In the graph above there is actually just one event for each entry, but it is stretched out until a new one arrives. Only thing I can think of is to send 0 every 30 seconds or so to all the labels.

**Comment by roidelapluie** [comment]
you could use   `some_metric and on (instance,job) push_time_seconds >= time()-(${__interval_ms}/1000)` in grafana.

push_time_seconds is updated on every push. With `push_time_seconds >= time()-(${__interval_ms}/1000)`, you should only see the value if it is present in the current interval chosen by grafana.

In general however, Prometheus is not an event based system. This kind of usecases is not fully supported.
